pkg/scheduler/framework/runtime/framework.go

```diff
@@ -352,6 +336,60 @@ func NewFramework(r Registry, profile *config.KubeSchedulerProfile, opts ...Opti
 		}
 	}
 
+	// initialize MultiPoint plugins
+	for _, e := range f.getExtensionPoints(profile.Plugins) {
+		plugins := reflect.ValueOf(e.slicePtr).Elem()
+		pluginType := plugins.Type().Elem()
+		// build enabledSet of plugins already registered via normal extension points
+		// to check double registration
+		enabledSet := sets.NewString()
+		for _, plugin := range e.plugins.Enabled {
+			enabledSet.Insert(plugin.Name)
+		}
+
+		disabledPlugins := sets.NewString()
+		for _, disabledPlugin := range e.plugins.Disabled {
+			disabledPlugins.Insert(disabledPlugin.Name)
+		}
+
+		for _, ep := range profile.Plugins.MultiPoint.Enabled {
+			pg, ok := pluginsMap[ep.Name]
+			if !reflect.TypeOf(pg).Implements(pluginType) {
+				continue
+			}
+
+			// a plugin that's enabled via MultiPoint can still be disabled for specific extension points
+			if disabledPlugins.Has(ep.Name) || disabledPlugins.Has("*") {
+				klog.V(4).InfoS("plugin disabled for extension point", "plugin", ep.Name, "extension", pluginType)
+				continue
+			}
+
+			if !ok {
+				return nil, fmt.Errorf("%s %q does not exist", pluginType.Name(), ep.Name)
+			}
+
+			if enabledSet.Has(ep.Name) {
+				return nil, fmt.Errorf("plugin %q already registered as %q", ep.Name, pluginType.Name())
```

Comment:
It can be enabled, but you have to disable it first. That's the current behavior if you want to modify default weight right? Or did we change that?

Reply:
We changed that

Reply:
ok, lets keep it that way then for Score.

Reply:
true, Score is the only case where it matters. Do you have a link to the changes? I remember talking about that issue with disabling/reenabling for weights but can't find the PR.

Reply:
Updated this, I realized that we only need to check double-registration for MultiPoint plugins against other MultiPoint plugins. This is because we are choosing to give explicit, individual extensions precedence. So by keeping track of the individual plugins (already registered above this point) separate from MultiPoint, we can handle that error by just skipping, exactly how we do with the default plugins now.